src/routes/productRoute.ts

```diff
@@ -0,0 +1,28 @@
+import express from 'express'
+import configureMulter from '../middlewares/multer'
+import validateRequest from '../utils/validateRequest'
+import productSchema from '../validations/productValidation'
+import collectionSchema from '../validations/collectionValidation'
+import isAuthenticated, {
+  checkPermission,
+} from '../middlewares/authenticationMiddleware'
+import productController from '../controllers/productController'
+import { UserRole } from '../database/models/userModel'
+
+const router = express.Router()
+
+router.post(
+  '/item',
+  isAuthenticated,
+  checkPermission(UserRole.SELLER),
+  configureMulter,
+  validateRequest(productSchema, 'body'),
+  productController.addProduct,
+)
```

Comment:
```suggestion
router.post(
  '/collections/:collectionId/items',
  isAuthenticated,
  checkPermission(UserRole.SELLER),
  configureMulter,
  validateRequest(productSchema, 'body'),
  productController.addProduct,
)
```
it would better if the route is descriptive